### ISSUE TITLE
Recent aws_cloudfront_distribution change breaks a usage pattern

### ISSUE THREAD

**Issue by lievertz** [terraform-providers/terraform-provider-aws#7773]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

TERRAFORM_VERSION=0.11.11

### Affected Resource(s)

* aws_cloudfront_distribution

### Terraform Configuration Files

Relevant bit:
```hcl
  viewer_certificate {
    acm_certificate_arn             = "${var.acm_certificate_arn}"
    cloudfront_default_certificate  = "${var.acm_certificate_arn == "" ? true : false}"
    ssl_support_method              = "sni-only"
  }
```
Also see similar example: https://github.com/cloudposse/terraform-aws-cloudfront-cdn/blob/master/main.tf

### Debug Output

This is a pretty straightforward situation, so omitting this step.

### Panic Output

N/A

### Expected Behavior

Provider users were previously specifying the acm_certificate_arn and the cloudfront_default_certificate where the default option would use a ternary to only show true if acm_certificate_arn was blank.

### Actual Behavior

Now you guys are explicitly enforcing only using one of those options.

### Steps to Reproduce

Use both acm_certificate_arn as well as cloudfront_default_certificate in viewer_certificate. Use normally.

### References

* https://github.com/terraform-providers/terraform-provider-aws/commit/3dc190af0e9828b8a641749b5041c23d6b741240
* Specifically at: https://github.com/terraform-providers/terraform-provider-aws/blame/master/aws/resource_aws_cloudfront_distribution.go#L611

Mostly I just wanted to file this to make sure you guys were aware that this was going to break some, perhaps a non-trivial number, of users... technically it matches the documentation (i.e., https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html#viewer-certificate-arguments), so feel free to close as FAD... but... thought you might want a heads up... and I think the nature of this commit could potentially have introduced other issues as well... so again, just a heads up in case any of this is surprising or undesirable.

**Comment by bflad** [comment]
Hi @lievertz 👋 Thanks for writing in.

Yesterday we released [version 2.0.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#200-february-27-2019), which includes this resource schema change. Originally the code _intended_ to include this `ConflictsWith` statement, but it was [errantly written in the original implementation](https://github.com/terraform-providers/terraform-provider-aws/pull/7732/files#diff-a9d90d298f8013ea41caa2d4f3b886a2L627) and it did not work as expected. During our major version upgrade work, this was noticed and fixed from the Terraform AWS Provider perspective.

The inclusion of the fix was actually stemming from the upcoming [Terraform 0.12 ability to also support `null` values in arguments](https://www.hashicorp.com/blog/terraform-0-12-conditional-operator-improvements), which the upstream Terraform Module would be expected to properly implement in this case when Terraform 0.12 is more generally available, e.g. something like

```hcl
# variables.tf
variable "acm_certificate_arn" {
  description = "Existing ACM Certificate ARN"
  default     = null
  type        = string
}

# main.tf
resource "aws_cloudfront_distribution" "default" {
  # ... other configuration ...

  viewer_certificate {
    acm_certificate_arn            = var.acm_certificate_arn
    ssl_support_method             = "sni-only"
    minimum_protocol_version       = var.viewer_minimum_protocol_version
    cloudfront_default_certificate = var.acm_certificate_arn ? null : true
  }
```

However, given the Terraform 0.12 release timing and longer tail for general Terraform 0.12 adoption, I believe we would be amenable to temporarily remove the "fixed" `ConflictsWith` usage in this point and defer its usage until another future major version release of the Terraform AWS Provider.

I'll mark this issue for our current release milestone (releasing next week) and ensure it is triaged.

If you need to [pin to the last 1.X Terraform AWS Provider release](https://www.terraform.io/docs/providers/aws/guides/version-2-upgrade.html#provider-version-configuration), a configuration like this can do that in the meantime:

```hcl
provider "aws" {
  # ... other configuration ...

  version = "~> 1.60"
}
```

**Comment by lievertz** [comment]
Hi @bflad Thanks for the response! Yeah... we appear to have stumbled kind of wholesale into 2.0 unawares, so thank you for for the context... sorry I missed that searching for a more specific issue -- I kind of missed the larger boat there!

Thank you for looking at that hold-off for implementing that validation as it will be nice to approach the later fix with a direct replacement.

Much appreciated!

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.1.0 of the Terraform AWS Provider, likely middle of this week. 👍

**Comment by rburke789** [comment]
Does anyone have a workaround for this until 2.1.0 is available? I am kinda blocked by this

**Comment by bflad** [comment]
This has been released in [version 2.1.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#210-march-07-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by rburke789** [comment]
awesome. Problem solved! Thanks!
